File: src/main/java/com/logics/logics/config/WebSocketServerConfig.java
package com.logics.logics.config;

import java.util.Map;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.HandlerMapping;
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping;
import reactor.core.publisher.Hooks;

@Configuration
@Slf4j
public class WebSocketServerConfig {
  @Bean
  public HandlerMapping webSocketMapping(ChatWebSocketHandler webSocketHandler) {
    return new SimpleUrlHandlerMapping(Map.of("/chat", webSocketHandler), -1);
  }

}

File: src/main/java/com/logics/logics/config/WebSocketServerConfig.kt
package com.logics.logics.config

import com.logics.logics.handlers.GameWebSocketHandler
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.HandlerMapping
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping

@Configuration
open class WebSocketServerConfig {

    private val logger = LoggerFactory.getLogger(WebSocketServerConfig::class.java)

    @Bean
    open fun webSocketMapping(
        chatWebSocketHandler: ChatWebSocketHandler,
        roomWebSocketHandler: RoomWebSocketHandler,
        playerWebSocketHandler: PlayerWebSocketHandler,
        gameWebSocketHandler: GameWebSocketHandler,
        gameChatWebSocketHandler: GameChatWebSocketHandler,
        questWebSocketHandler: QuestWebSocketHandler
    ): HandlerMapping {
        logger.info("Configuring WebSocket mappings")
        return SimpleUrlHandlerMapping(
            mapOf(
                "/chat" to chatWebSocketHandler,
                "/rooms" to roomWebSocketHandler,
                "/players" to playerWebSocketHandler,
                "/games" to gameWebSocketHandler,
                "/gamechat" to gameChatWebSocketHandler,
                "/quest" to questWebSocketHandler
            ),
            -1
        )
    }
}